app/helpers/administrate/application_helper.rb

```diff
@@ -15,6 +15,20 @@ def render_field(field, locals = {})
       render locals: locals, partial: field.to_partial_path
     end
 
+    def requireness(field)
+      required_field?(field) ? "required" : "optional"
+    end
+
+    def required_field?(field)
+      has_presence_validator?(field.resource.class, field.attribute)
+    end
+
+    def has_presence_validator?(resource_class, field_name)
+      resource_class.validators_on(field_name).
```

Comment:
What do you think about doing only one iteration over these validators?
```ruby
def has_presence_validator?(resource_class, field_name)
  resource_class
    .validators_on(field_name)
    .any? { |validator| validator.class == ActiveRecord::Validations::PresenceValidator }
end
```

Reply:
Nice!

Reply:
this line will exceed line size:
```ruby
any? { |validator| validator.class == ActiveRecord::Validations::PresenceValidator }
```

do you think that would be ok?
```ruby
any? { |v| v.class == ActiveRecord::Validations::PresenceValidator }
```

Reply:
In such a small context it is fine.